selftests/check.py

```diff
@@ -743,16 +748,35 @@ def create_suites(args):  # pylint: disable=W0621
 
         suites.append(TestSuite.from_config(config_check_optional, "optional-plugins"))
 
+    test_dir = "vmimage"
+
+    # Combined vmimage option - tests first, then variants
     if args.dict_tests.get("vmimage"):
-        test_dir = os.path.join("selftests", "pre_release", "tests")
-        vmimage_config = {
-            "resolver.references": [os.path.join(test_dir, "vmimage.py")],
+        # First suite: vmimage tests
+        vmimage_tests_config = {
```

Comment:
IMO these changes are related to the vmimage testing refactor in 7937e42ad6470c96a077bfedf2a7cd9b31eeae2 , am I right? Therefore, I would move it to that commit. 

TIP:
The best way how to separate those changes is with `git add -p` command.